### ISSUE TITLE
Doesn't honor 3DS' proxy settings

### ISSUE THREAD

**Issue by rctgamer3** [Wolvan/Homebr3w#24]
I have all 3DS data routed through a local proxy to control what my 3DS can access or not. However, Homebr3w doesn't check at all for a set proxy.

**Comment by Wolvan** [comment]
Possibly lpp-3ds' fault. Is it like this for any lpp apps, could you test this? @Rinnegatamante, does LPP-3ds use a system defined proxy at all or do I have to set something for that to happen manually?

**Comment by Rinnegatamante** [comment]
At least when i used to update lpp-3ds, ctrulib didn't have any function related to proxies.

**Comment by Wolvan** [comment]
what about that https://github.com/smealum/ctrulib/blob/master/libctru/source/services/httpc.c#L90
Could that be a thing that gives you proxy access?

**Comment by Rinnegatamante** [comment]
Possibly but i don't have enough time to give it a try.

**Comment by Wolvan** [comment]
Calling @ksanislo into this for his httpc knowledge

**Comment by ksanislo** [comment]
@Wolvan & @Rinnegatamante It looks like the 0 for the use proxy bool is hardcoded here at https://github.com/Rinnegatamante/lpp-3ds/blob/master/source/luaNetwork.cpp#L140 to false... You can probably just switch it to 1 and recompile and see if that does the trick. I'm not sure if there's some mechanism to check for a default proxy before you attempt to enable it though. So it may need some kind of check to get the true/false value needed first.

**Comment by Rinnegatamante** [comment]
I can provide a test version with it set to 1 (true) if needed. Let me know @Wolvan

**Comment by Wolvan** [comment]
That would be very much appreciated, thanks Rin.

**Comment by Rinnegatamante** [comment]
https://www.rinnegatamante.it/lpp-3ds_proxy.3dsx

**Comment by Wolvan** [comment]
Can I get the elf file perhaps?

**Comment by Rinnegatamante** [comment]
https://www.rinnegatamante.it/lpp-3ds_proxy.elf

**Comment by Wolvan** [comment]
@rctgamer3 can you try https://a.pomf.cat/svuipe.cia ? Just build that with Rin's proxy version

**Comment by rctgamer3** [comment]
Not sure if related, but (with that .cia) i get `Error: [string"?"]:568: bad argument #3 to 'requestString' (number expected, got string)` on startup after the loading config message.

**Comment by Wolvan** [comment]
You do have a proxy set though, right?

**Comment by rctgamer3** [comment]
Yes, I do have a proxy set, I haven't changed/disabled my proxy settings in a long time (set to my computer's DNS address and port 8888). No traffic shows up.

**Comment by Rinnegatamante** [comment]
According to ctrulib header file ( https://github.com/smealum/ctrulib/blob/master/libctru/include/3ds/services/httpc.h#L53 ), use_defaultproxy should be set to 0 to use default proxy (so standard lpp-3ds should already use default proxy in system settings) :/

**Comment by ksanislo** [comment]
I'm pretty sure that header comment is wrong then. Because this is what it really does in the ctrulib source:


	if(use_defaultproxy==0)return 0;

	ret = HTTPC_SetProxyDefault(context->servhandle, context->httphandle);

**Comment by rctgamer3** [comment]
Bump.

**Comment by rctgamer3** [comment]
Bump.

**Comment by rctgamer3** [comment]
Bump
